test/spmd/tensor/test_ops.py

```diff
@@ -102,6 +103,43 @@ def test_ones_like(self):
         ones_expected = torch.ones(4, 8)
         self.assertEqual(ones_expected, ones_like_dt.local_tensor())
 
+    def _run_sharded_elementwise_ops(
+        self, mesh, spec, input_size, op, **kwargs
+    ):
+        torch.manual_seed(self.rank)
+        input_tensor = torch.randn(*input_size, requires_grad=True)
+        dist_tensor = Tensor.from_local(input_tensor, mesh, spec)
+        dt = op(dist_tensor, **kwargs)
+        expected = op(input_tensor, **kwargs)
+        self.assertEqual(input_tensor, dist_tensor.local_tensor())
+        if "training" in kwargs and kwargs["training"]:
+            self.assertNotEqual(expected, dt.local_tensor())
+        else:
+            self.assertEqual(expected, dt.local_tensor())
+
+    @with_comms
+    def test_activations(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        self._run_sharded_elementwise_ops(device_mesh, [Shard(0)], (8, 5), torch.nn.functional.gelu)
+        self._run_sharded_elementwise_ops(device_mesh, [Replicate()], (8, 5), torch.nn.functional.gelu)
+        self._run_sharded_elementwise_ops(device_mesh, [Shard(1)], (3, 14), torch.nn.functional.relu)
+        self._run_sharded_elementwise_ops(device_mesh, [Replicate()], (8, 5), torch.nn.functional.relu)
+
+    @with_comms
+    def test_dropout(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        self._run_sharded_elementwise_ops(device_mesh, [Shard(0)], (8, 5),
+                                          torch.nn.functional.dropout, p=0.4, training=False)
+        self._run_sharded_elementwise_ops(device_mesh, [Shard(1)], (3, 14),
+                                          torch.nn.functional.dropout, p=0.5, training=True)
+
+    @with_comms
+    def test_dropout_errors(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        with self.assertRaisesRegex(RuntimeError, 'Not supported!'):
+            self._run_sharded_elementwise_ops(device_mesh, [_Partial(ReduceOp.SUM)], (8, 5),
```

Comment:
@wanchaol Looks like partial tensor does not trigger the override. So no errors are thrown here.